### PR TITLE
Models Repository Refactor

### PR DESCRIPTION
**Pull request by vilit1**
Refactor with respect to track 2.

image

@digimaun

### CODE REVIEW

**Comment by vilit1** [comment]
Local IT with different (clean) environments:
python 3.6:
image

python 3.7
image

python 3.8
image

python 3.9
image

File: sdk/modelsrepository/azure-iot-modelsrepository/samples/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This directory contains samples showing how to use the features of the Azure IoT Models Repository Library.
 
-The pre-configured endpoints and DTMIs within the samples refer to example models that can be found on [devicemodels.azure.com](https://devicemodels.azure.com/). These values can be replaced to reflect the locations of your own models, wherever they may be.
+The pre-configured endpoints and DTMIs within the samples refer to example models that can be found on [devicemodels.azure.com](https://devicemodels.azure.com/) or in the provided local repository [SampleModelRepo](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/modelsrepository/azure-iot-modelsrepository/samples/SampleModelRepo/). These values can be replaced to reflect the locations of your own models, wherever they may be.
```
Comment (digimaun) [inline sdk/modelsrepository/azure-iot-modelsrepository/samples/README.md:5]: This link looks off, the sample models repo directory is called `SamplesModelsRepo` not `SampleModelRepo`.

Comment (vilit1) [reply sdk/modelsrepository/azure-iot-modelsrepository/samples/README.md:5]: This link won't exist till the pr is merged in

Comment (digimaun) [reply sdk/modelsrepository/azure-iot-modelsrepository/samples/README.md]: The pipeline appears to resolve the link with respect to the latest PR commit

**Pull request closed without merging**